src/NServiceBus.RabbitMQ.AcceptanceTests/DelayedDelivery/When_Deferring_a_message_longer_than_allowed_maximum.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace NServiceBus.AcceptanceTests.DelayedDelivery
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using Features;
+    using NUnit.Framework;
+
+    public class When_Deferring_a_message_longer_than_allowed_maximum : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public void Should_throw()
+        {
+            var delay = TimeSpan.FromDays(365 * 10);
```

Comment:
Can we use `TimeSpan.MaxValue` in this place. As current code assumes that we are using 28 bits as a delay level which translates to more than 8 years. This value is to big as it represents 10 years. But if we would like to increase 28 bit level to something higher we need to remember to modify this test as well. I would suggest to either take max value, or to somehow expose the bit level value so that we can refer to it in tests.